tests/gpflow/models/test_gpr.py

```diff
@@ -45,3 +46,20 @@ def test_non_trainable_model_objective():
 
     _ = model.log_marginal_likelihood()
     assert model.log_prior_density() == 0.0
+
+
+def test_varying_data():
+    input_dim = 2
+    output_dim = 1
+    N = 5
+    X, Y = rng.randn(N, input_dim), rng.randn(N, output_dim)
+    var_data = (tf.Variable(X, shape=[None, input_dim]), tf.Variable(Y, shape=[None, output_dim]))
+    m = gpflow.models.GPR(var_data, gpflow.kernels.SquaredExponential())
+    lml_func = tf.function(m.log_marginal_likelihood)
+    old_lml = lml_func()
+    new_N = 7
+    new_X, new_Y = rng.randn(new_N, input_dim), rng.randn(new_N, output_dim)
+    var_data[0].assign(new_X)
+    var_data[1].assign(new_Y)
+    new_lml = lml_func()
+    assert np.abs((old_lml - new_lml) / (new_lml + old_lml)) > 0.1
```

Comment:
what does this .1 mean?

Reply:
it means "their relative difference is larger than 10%"

Reply:
(well, 5% technically, there's a factor of 2 missing)

Reply:
I'll add a comment clarifying that this is asserting they're *different*